test-cases/jetpack/story.md

```diff
@@ -0,0 +1,88 @@
+# Story Block - Test Cases
+
+--------------------------------------------------------------------------------
+
+##### TC001
+
+#### Precondition
+
+A wordpress.com site or a jetpack site with jetpack version 9.1 or newer
+
+### Block picker shows the Story block
+
+1. On the site described above, add a post, then tap on + to open the Block Picker
+2. Verify the Story block appears among the blocks that can be added to the Post, and add it
+3. An empty Story block should have been added to the Post (with the Add Media placeholder)
+
+
+##### TC002
+
+#### Precondition
+
+A. A jetpack site with jetpack version previous to 9.1
+
+B. A self-hosted site without jetpack
+
+### Block picker does not show the Story block
+
+1. On the site described above, add a post, then tap on + to open the Block Picker
+2. Verify the Story block does NOT appear among the blocks that can be added to the Post.
+3. Perform steps 1 and 2 on a self-hosted site (option B above)

```

Comment:
I'm torn on this test case mostly because it would require some setup and will gradually become obsolete as people upgrade. But if this has dependencies outside Gutenberg-Mobile, or if we think the check might end up getting removed prematurely, then I'm ok keeping it around. 

I'm suggesting a change yet; I mostly wanted to voice my thoughts and get your input.

Reply:
Good observation @chipsnyder !

> [...] will gradually become obsolete as people upgrade

The Story block is a .com / jetpack block so there's no way a pure self-hosted site will have the Story block. Hence the preparation for the test (a jetpack site _and_ a self hosted site at least) and then actually testing for presence / absence of the Story block as an option in the block picker depending on whether this is a jetpack / pure self-hosted site needs to be done. This specially since we have ran into trouble in the past, that's why I added this test specifically 😅 

> But if this has dependencies outside Gutenberg-Mobile

Exactly :) 👍